lib/compat/wordpress-6.1/date-settings.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Updates to the settings given to @wordpress/date.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Removes the call to wp.date.setSettings() added by Core and adds our own call
+ * to wp.date.setSettings().
+ *
+ * This lets us add a new l10n.dayOfWeek setting.
+ *
+ * To merge this into Core, simply update the wp.date.setSettings() call in
+ * wp_default_packages_inline_scripts.
+ *
+ * @param WP_Scripts $scripts WP_Scripts object.
+ */
+function gutenberg_update_date_settings( $scripts ) {
+	global $wp_locale;
+
+	$inline_scripts = $scripts->get_data( 'wp-date', 'after' );
+	if ( $inline_scripts ) {
+		foreach ( $inline_scripts as $index => $inline_script ) {
+			if ( str_starts_with( $inline_script, 'wp.date.setSettings' ) ) {
+				unset( $scripts->registered['wp-date']->extra['after'][ $index ] );
+			}
+		}
+	}
+
+	// Calculate the timezone abbr (EDT, PST) if possible.
+	$timezone_string = get_option( 'timezone_string', 'UTC' );
+	$timezone_abbr   = '';
+
+	if ( ! empty( $timezone_string ) ) {
+		$timezone_date = new DateTime( 'now', new DateTimeZone( $timezone_string ) );
+		$timezone_abbr = $timezone_date->format( 'T' );
+	}
+
+	$scripts->add_inline_script(
+		'wp-date',
+		sprintf(
+			'wp.date.setSettings( %s );',
+			wp_json_encode(
+				array(
+					'l10n'     => array(
+						'locale'        => get_user_locale(),
+						'months'        => array_values( $wp_locale->month ),
+						'monthsShort'   => array_values( $wp_locale->month_abbrev ),
+						'weekdays'      => array_values( $wp_locale->weekday ),
+						'weekdaysShort' => array_values( $wp_locale->weekday_abbrev ),
+						'meridiem'      => (object) $wp_locale->meridiem,
+						'relative'      => array(
+							/* translators: %s: Duration. */
+							'future' => __( '%s from now', 'gutenberg' ),
+							/* translators: %s: Duration. */
+							'past'   => __( '%s ago', 'gutenberg' ),
+						),
+						'startOfWeek'   => (int) get_option( 'start_of_week', 0 ),
```

Comment:
The change in this file looks good to me, just confirming that the only "real" difference between core and this PR (in terms of this added script) is this line adding `startOfWeek`?

Reply:
Yes correct! The rest is painful duplication because there's no better way to extend this. Probably we should add a filter!